src/replicache.ts

```diff
@@ -453,40 +462,75 @@ export class Replicache<MD extends MutatorDefs = {}>
     return res.root;
   }
 
-  private _setRoot(root: Promise<string | undefined>) {
-    this._root = root;
-    this._setStorage(root);
-  }
+  private _onStorage = (e: StorageEvent) => {
+    const {key, newValue} = e;
+    if (newValue && key === storageKeyName(this._name)) {
+      const {root, changedKeys, index} = JSON.parse(newValue) as {
```

Comment:
Maybe define this type:

```
type StorageBroadcastData = BroadcastData & {changedKeys: [string, string[]][]};
```

... that way you can also make the assignment to local storage typesafe.